commons-db/src/main/java/com/palantir/nexus/db/pool/RetriableTransactions.java

```diff
@@ -0,0 +1,348 @@
+                        result = newResult;
+                        c.commit();
+
+                        // if we got here we'll return even if cleanTxLog() throws
+                        ret = true;
+
+                        cleanTxLog(c, id);
+                        c.commit();
+
+                        return;
+                    } finally {
+                        c.close();
+                    }
+                } catch (SQLException e) {
+                    if (ret) {
+                        squelch(e);
+                        return;
+                    }
+                    throw e;
+                }
+            }
+
+            private boolean attemptVerify() throws SQLException {
+                boolean ret = false;
+                try {
+                    Connection c = cm.getConnection();
+                    try {
+                        ret = checkTxLog(c, id);
+                        if (ret) {
+                            cleanTxLog(c, id);
+                        }
+                        return ret;
+                    } finally {
+                        c.close();
+                    }
+                } catch (SQLException e) {
+                    if (ret) {
+                        // Once we know we're in we may or may not have tried to delete
+                        // leaving us the sole carrier of that tx having completed so
+                        // we can't throw.
+                        squelch(e);
+                        return ret;
+                    }
+                    throw e;
+                }
+            }
+
+            private void squelch(SQLException e) {
+                log.warn("Squelching SQLException while trying to clean up retriable write transaction id " + id, e);
+            }
+        }
+        return new LexicalHelper().run();
+    }
+
+    private static final LoadingCache<ConnectionManager, AtomicBoolean> createdTxTables = CacheBuilder.newBuilder().weakKeys().build(new CacheLoader<ConnectionManager, AtomicBoolean>() {
+        @Override
+        public AtomicBoolean load(ConnectionManager cm) {
+            return new AtomicBoolean(false);
+        }
+    });
+    private static void createTxTable(ConnectionManager cm) throws SQLException {
+        AtomicBoolean createdTxTable = createdTxTables.getUnchecked(cm);
+        if (createdTxTable.get()) {
+            return;
+        }
+
+        String varcharType = null;
+        switch (cm.getDbType()) {
+            case ORACLE:
+                varcharType = "VARCHAR2";
+                break;
+
+            case POSTGRESQL:
+            case H2_MEMORY:
+                varcharType = "VARCHAR";
+                break;
+
+            default:
+                throw new IllegalStateException();
+        }
+        if (varcharType == null) {
+            throw new IllegalStateException();
+        }
+
+        // I'd love to CREATE TABLE ... IF NOT EXISTS but Ye Olde Postgres (TM)
```

Comment:
clean this up and use create table if not exists, and take a strict dependency on postgres 9.1+